trio/_core/tests/tutil.py

```diff
@@ -80,6 +81,40 @@ def ignore_coroutine_never_awaited_warnings():
             gc_collect_harder()
 
 
+def _noop(*args, **kwargs):
+    pass
+
+
+if sys.version_info >= (3, 8):
+
+    @contextmanager
+    def restore_unraisablehook():
+        sys.unraisablehook, prev = sys.__unraisablehook__, sys.unraisablehook
+        try:
+            yield
+        finally:
+            sys.unraisablehook = prev
+
+    @contextmanager
+    def disable_threading_excepthook():
+        threading.excepthook, prev = _noop, threading.excepthook
```

Comment:
sadly `threading.__excepthook__` is only in 3.10

Reply:
Probably shouldn't hold this up.  Maybe I'll add something in https://github.com/python-trio/trio/pull/1921.